borsh-rs/borsh/src/ser/mod.rs

```diff
@@ -122,12 +123,17 @@ where
     }
 }
 
+impl BorshSerialize for str {
+    #[inline]
+    fn serialize<W: Write>(&self, writer: &mut W) -> io::Result<()> {
+        self.as_bytes().serialize(writer)
+    }
+}
+
```

Comment:
Please add borshdeserialize for str as well (if this makes sense)

Reply:
There is no sense in that since `str` is unsized, you cannot deserialize into it; concrete example:

```rust
#[derive(BorshSerialize)]
struct Serializable<'a> {
    s: &'a str,
}

// This does not work since there is no owner for the `&str` data.
//#[derive(BorshDeserialize)]
//struct NonDeserializable<'a> {
//   s: &'a str,
//}

// However, Cow<'a, str> and Box<str> are both serializable and deserializable since Cow::Owned and Box can own the data.
#[derive(BorshSerialize)]
struct Both<'a> {
    c: Cow<'a, str>,
    b: Box<str>,
}
```